packages/block-editor/src/store/selectors.js

```diff
@@ -1284,6 +1284,45 @@ export const hasInserterItems = createSelector(
 	],
 );
 
+/**
+ * Determines whether there is only one item that may be inserted.
+ *
+ * @param {Object}  state        Editor state.
+ * @param {?string} rootClientId Optional root client ID of block list.
+ *
+ * @return {boolean} True if there is one item available, false if zero or more than one.
+ */
+export const __experimentalHasOnlyOneAllowedInserterItem = ( state, rootClientId = null ) => {
+	if ( rootClientId ) {
+		const parentBlockListSettings = getBlockListSettings( state, rootClientId );
+		return get( parentBlockListSettings, [ 'allowedBlocks', 'length' ], 0 ) === 1;
+	}
+
+	return false;
+};
+
+/**
+ * Gets the name of the only item that may be inserted.
+ *
+ * @param {Object}  state        Editor state.
+ * @param {?string} rootClientId Optional root client ID of block list.
+ *
+ * @return {string?} The name of the allowed block or null.
+ */
+export const __experimentalGetTheOnlyAllowedItem = ( state, rootClientId = null ) => {
```

Comment:
```suggestion
export const __experimentalGetTheOnlyAllowedBlockType = ( state, rootClientId = null ) => {
```

Reply:
I renamed them both to contain BlockType instead of Item